[next] Implement filter settings (homepage)

andrewywong opened the pull request:
https://github.com/HolodexNet/Holodex/issues/706

Changes:
- Add the video list settings menu component to the homepage tabs list. 
  - Menu opens in a popover similar to the current site.
  - Menu component isn't shown when on the Members tab.
  - On the menu
    - Video filter fields are shown - the same fields are also found under the `Settings > Content Preferences > Filter Videos`
    - The "Sort By" select field & label is shown only when on the Live/Upcoming tab.
![Screenshot 2025-03-18 002036](https://github.com/user-attachments/assets/d3a7acca-4111-484d-9eee-c25ce3ae20c5)
    - The "Uploaded Before" date picker field & label is shown only when on the Archive or Clips tabs.
![Screenshot 2025-03-18 001837](https://github.com/user-attachments/assets/0ddd9235-07ea-46f2-9d69-93a5e1c48c85)
- Live/Upcoming tab uses the "Sort By" value to sort live streams.

https://github.com/user-attachments/assets/a3f8aff0-7de5-4368-a33b-57af11b35925

- Archive and Clips tabs use the "Uploaded Before" value to filter past videos.

https://github.com/user-attachments/assets/6b111b67-dd34-411d-8968-3f8690382444

- Filter videos on Live/Upcoming, Archive, and Clips tabs via the video list settings menu.

https://github.com/user-attachments/assets/f705c66e-4f16-4936-aeb0-54c0b7ef55c6

- New homepage filtering/sorting changes also apply to the favorites page.
- Small UI fixes and performance optimizations.

andrewywong commented:
I've been keeping a list of open questions and notes while working on the feature. Could I get some suggestions on how to address the listed items below? It would greatly help in refining the feature's direction and reducing ambiguity.

- Regarding the video list settings menu UI/UX, should we look to maintain design parity with the current site, and if so, to what extent? (I'm currently looking to bring it close while still abiding by the new site's design system.)
- Any preferences on how we should show the label for the "Sort by" select and "Uploaded Before" date picker? It looks like we no longer use floating labels on the new site, so the menu layout will look slightly different.
- The current date (time) picker component isn't set up to clear the selected date (either via close icon or unselecting the date on the calendar), and it always shows the time input below the calendar (rather than being a pure date picker). 
  - Should the date picker on the menu be able to clear the selected date (like on the current site), and if so, how should we clear it (clear button or/and unselecting date on calendar)? Also, should the menu date picker allow only for day selection (like on the current site) or also allow for time selection now? 
  - On a related note: how do we feel about refactoring the current `DatePicker `component into a `DateTimePicker `and creating a new separate `DatePicker `component to use on the menu?

- It seems like we have filtering logic for ignored topics in `useVideoFilter`. However, since the "Ignored Topics" input component hasn't been implemented yet, and since it looks to be a shared feature between the settings and homepage, (in which we could create a component to be reused across both pages), I'll treat that as a separate "todo" feature outside the scope of the filter settings menu.

- How do we feel about the naming and structure of the new components, atoms, and files (particularly for the video list settings menu component)?
- Is it fine for the settings menu component to reuse the `ToggleableFeatureGroup` component or should we create another layout component to display the `features` (Filter Videos) fields?

sphinxrave commented:
Wow. Thanks for the detailed PR! It's very well crafted.



> Regarding the video list settings menu UI/UX, should we look to maintain design parity with the current site, and if so, to what extent? (I'm currently looking to bring it close while still abiding by the new site's design system.)

I think you're maintaining a healthy balance here!


> Any preferences on how we should show the label for the "Sort by" select and "Uploaded Before" date picker? It looks like we no longer use floating labels on the new site, so the menu layout will look slightly different.

Right, we currently don't have a clear-set design for it b/c shadcn doesn't really specify one either... I think we're just winging it everywhere atm (sorry).


> The current date (time) picker component isn't set up to clear the selected date (either via close icon or unselecting the date on the calendar), and it always shows the time input below the calendar (rather than being a pure date picker).
> Should the date picker on the menu be able to clear the selected date (like on the current site), and if so, how should we clear it (clear button or/and unselecting date on calendar)? Also, should the menu date picker allow only for day selection (like on the current site) or also allow for time selection now?
> On a related note: how do we feel about refactoring the current DatePicker component into a DateTimePicker and creating a new separate DatePicker component to use on the menu?

err, for clearing the value maybe a X button on the right of it, but outside the Picker component? 

The time input on the dropdown... let's add an optional `showTime` prop. maybe like `<DatePicker showTime={false}>`


> It seems like we have filtering logic for ignored topics in useVideoFilter. However, since the "Ignored Topics" input component hasn't been implemented yet, and since it looks to be a shared feature between the settings and homepage, (in which we could create a component to be reused across both pages), I'll treat that as a separate "todo" feature outside the scope of the filter settings menu.

That sounds good. 


> How do we feel about the naming and structure of the new components, atoms, and files (particularly for the video list settings menu component)?

It's okay 👍 

> Is it fine for the settings menu component to reuse the ToggleableFeatureGroup component or should we create another layout component to display the features (Filter Videos) fields?

`ToggleableFeatureGroup` looks fine but has some slight padding incongruence. I think lightly retouch it to bring it in line to what your dropdown is doing? Don't have to be this PR though.

andrewywong commented:
Thanks for the feedback! I've updated the PR and added a video in the description showcasing the new functionality to clear the date picker input. I'm too lazy to update the existing videos with the most recent UI, but if you want to see those as well, then please let me know.

sphinxrave commented:
Great work <3
